Source/monster.cpp

```diff
@@ -1215,12 +1199,8 @@ void SetMapMonsters(const uint16_t *dunData, Point startPosition)
 
 void DeleteMonster(int i)
 {
-	int temp;
-
 	nummonsters--;
-	temp = monstactive[nummonsters];
-	monstactive[nummonsters] = monstactive[i];
-	monstactive[i] = temp;
+	std::swap(monstactive[nummonsters], monstactive[i]);

```

Comment:
I'm curious if the swap is really necessary. Seems like you could just do `monstactive[i] = monstactive[nummonsters];`.